docs/book/src/providers/implementers-guide/controllers_and_reconciliation.md

```diff
@@ -0,0 +1,254 @@
+# Controllers and Reconciliation
+
+From the [kubebuilder book][controller]:
+
+> Controllers are the core of Kubernetes, and of any operator.
+>
+> It’s a controller’s job to ensure that, for any given object, the actual state of the world (both the cluster state, and potentially external state like running containers for Kubelet or loadbalancers for a cloud provider) matches the desired state in the object. 
+> Each controller focuses on one root Kind, but may interact with other Kinds.
+>
+> We call this process reconciling.
+
+[controller]: https://book.kubebuilder.io/cronjob-tutorial/controller-overview.html#whats-in-a-controller
+
+Right now, we can create objects in our API but we won't do anything about it. Let's fix that. 
+
+# Let's see the Code
+
+Kubebuilder has created our first controller in `controllers/mailguncluster_controller.go`. Let's take a look at what got generated:
+
+```go
+// MailgunClusterReconciler reconciles a MailgunCluster object
+type MailgunClusterReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=mailgunclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=mailgunclusters/status,verbs=get;update;patch
+
+func (r *MailgunClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	_ = r.Log.WithValues("mailguncluster", req.NamespacedName)
+
+	// your logic here
+
+	return ctrl.Result{}, nil
+}
+```
+
+## RBAC Roles
+
+Those `// +kubebuilder...` lines tell kubebuilder to generate [RBAC] roles so the manager we're writing can access its own managed resources. 
+We also need to add roles that will let it retrieve (but not modify) Cluster API objects.
+So we'll add another annotation for that:
+
+```
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=mailgunclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=mailgunclusters/status,verbs=get;update;patch
```

Comment:
Maybe remove these 2 lines and only show the new annotation?

Reply:
I wanted the context so it's obvious where this line is supposed to go.

Reply:
Fair, thanks for the explanation